examples/chip-tool/commands/clusters/Commands.h

```diff
@@ -21,6 +21,369 @@
+| * DiscoverAttributesResponse                                          | 0x0D |
+\*----------------------------------------------------------------------------*/
+
+/*
+ * Command ReadAttributesResponse
+ */
+class ReadAttributesResponse : public ModelCommandResponse
+{
+public:
+    ReadAttributesResponse() : ModelCommandResponse(0x01) {}
+
+    bool HandleResponse(uint8_t * message, uint16_t messageLen) const override
+    {
+        ChipLogProgress(chipTool, "ReadAttributesResponse (0x01):");
+        // struct readAttributeResponseRecord[]
+        uint8_t * messageEnd = message + messageLen;
+        do
+        {
+            ChipLogProgress(chipTool, "  %s: 0x%04x", "attributeId", chip::Encoding::LittleEndian::Read16(message)); // attribId
+            ChipLogProgress(chipTool, "  %s: 0x%02x", "status", chip::Encoding::Read8(message));                     // zclStatus
+            ReadAnyType(message);
+        } while (message < messageEnd);
+
+        return true;
+    }
+};
+
+/*
+ * Command WriteAttributesResponse
+ */
+class WriteAttributesResponse : public ModelCommandResponse
+{
+public:
+    WriteAttributesResponse() : ModelCommandResponse(0x04) {}
+
+    bool HandleResponse(uint8_t * message, uint16_t messageLen) const override
+    {
+        ChipLogProgress(chipTool, "WriteAttributesResponse (0x04):");
+        // struct writeAttributeResponseRecord[]
+        uint8_t * messageEnd = message + messageLen;
+        do
+        {
+            ChipLogProgress(chipTool, "  %s: 0x%02x", "status", chip::Encoding::Read8(message));                     // zclStatus
+            ChipLogProgress(chipTool, "  %s: 0x%04x", "attributeId", chip::Encoding::LittleEndian::Read16(message)); // attribId
+        } while (message < messageEnd);
+
+        return true;
+    }
+};
+
+/*
+ * Command WriteAttributesNoResponse
+ */
+class WriteAttributesNoResponse : public ModelCommandResponse
+{
+public:
+    WriteAttributesNoResponse() : ModelCommandResponse(0x05) {}
+
+    bool HandleResponse(uint8_t * message, uint16_t messageLen) const override
+    {
+        ChipLogProgress(chipTool, "WriteAttributesNoResponse (0x05):");
+        // struct writeAttributeRecord[]
+        uint8_t * messageEnd = message + messageLen;
+        do
+        {
+            ChipLogProgress(chipTool, "  %s: 0x%04x", "attributeId", chip::Encoding::LittleEndian::Read16(message)); // attribId
+            ReadAnyType(message);
+        } while (message < messageEnd);
+
+        return true;
+    }
+};
+
+/*
+ * Command DefaultResponse
+ */
+class DefaultResponse : public ModelCommandResponse
+{
+public:
+    DefaultResponse() : ModelCommandResponse(0x0B) {}
+
+    bool HandleResponse(uint8_t * message, uint16_t messageLen) const override
+    {
+        ChipLogProgress(chipTool, "DefaultResponse (0x0B):");
+        ChipLogProgress(chipTool, "  %s: 0x%02x", "commandId", chip::Encoding::Read8(message));  // uint8
```

Comment:
How do we know this is not reading off the end of the buffer?  This seems to apply broadly to _all_ this code...